src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -862,11 +862,15 @@ private async Task<ConnectBlocksResult> ConnectBlockAsync(ChainedHeaderBlock blo
 
             if (validationContext.Error != null)
             {
-                List<int> badPeers;
+                var badPeers = new List<int>();
 
-                lock (this.peerLock)
+                // Ban the peers only in case block is invalid and not temporary rejected.
+                if (validationContext.RejectUntil == null)
```

Comment:
I see that we also set `RejectUntil ` in `PosFutureDriftRule`, so that means that if that rule fails we will also not ban the peer?

Reply:
Yes, correct